examples/connecting/sasl_ssl_plain.go

```diff
@@ -0,0 +1,49 @@
+package connect
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/twmb/kafka-go/pkg/kgo"
+	"github.com/twmb/kafka-go/pkg/sasl/plain"
+)
+
+func connectSaslSslPlain() {
+	fmt.Println("starting...")
+
+	seeds := []string{"localhost:9092"}
+	// SASL Plain credentials
+	user := ""
+	password := ""
+
+	opts := []kgo.Opt{
+		kgo.SeedBrokers(seeds...),
+
+		// SASL Options
+		kgo.SASL(plain.Plain(func(context.Context) (plain.Auth, error) {
+			return plain.Auth{
+				User: user,
+				Pass: password,
+			}, nil
+		})),
+
+		// Configure TLS. Uses SystemCertPool for RootCAs by default.
+		kgo.Dialer(func(ctx context.Context, host string) (net.Conn, error) {
+			return (tls.Dialer{
```

Comment:
Can this have an `&` before the tls.Dialer? `DialContext` requires a pointer receiver, so without the `&` this fails to compile if one just blindly copies and pastes this.

Reply:
Alternatively, this dialer could be defined above the options; as it currently exists, this does allocate a tls.Dialer and net.Dialer every dial. I think this is showing me that it's worth it to have a direct `kgo.TLSDialer` option, but I'm not sure what the args for that should be (maybe a tls.Config? maybe also a net.Dialer?)

Reply:
I've pushed a breaking API change for the Dialer, this new change should make this API slightly easier to use, and mirrors other places in the go stdlib that take dialers.